iis-common/src/main/java/eu/dnetlib/iis/common/report/PigCountersReportGenerator.java

```diff
@@ -0,0 +1,116 @@
+import eu.dnetlib.iis.common.schemas.ReportParam;
+
+/**
+ * Java workflow node process for building report from pig counters.<br/>
+ * <br/>
+ * It writes report properties into avro datastore of {@link ReportParam}s
+ * with location specified in output port.<br/>
+ * Report property key must start with <code>report.</code> to
+ * be included in output datastore.<br/>
+ * <br/>
+ * Report property values can contain placeholders for easier evaluation of
+ * pig counters. Placeholders are resolved using {@link PigCounterValueResolver}.<br/>
+ * <br/>
+ * Process needs <code>pigCounters</code> property that contains json representation
+ * of pig counters for working.<br/>
+ * 
+ * @author madryk
+ */
+public class PigCountersReportGenerator implements Process {
+
+    private static final String REPORT_PORT_OUT_NAME = "report";
+    
+    private static final String REPORT_PROPERTY_PREFIX = "report.";
+    
+    private static final String PIG_COUNTERS_PROPERTY = "pigCounters";
+    
+    
+    private PigCountersParser pigCountersParser = new PigCountersParser();
+    
+    private PigCounterValueResolver pigCounterValueResolver = new PigCounterValueResolver();
+    
+    
+    //------------------------ LOGIC --------------------------
+    
+    @Override
+    public Map<String, PortType> getInputPorts() {
+        return Collections.emptyMap();
+    }
+
+    @Override
+    public Map<String, PortType> getOutputPorts() {
+        return Collections.singletonMap(REPORT_PORT_OUT_NAME, new AvroPortType(ReportParam.SCHEMA$));
+    }
+    
+    @Override
+    public void run(PortBindings portBindings, Configuration conf, Map<String, String> parameters) throws Exception {
+        
+        String pigCountersJson = parameters.get(PIG_COUNTERS_PROPERTY);
+        
+        PigCounters pigCounters = pigCountersParser.parse(pigCountersJson);
+        
+        
+        Map<String, String> reportParams = collectReportParameters(parameters);
+        
+        List<ReportParam> avroReportParams = convertToAvroReportParams(reportParams, pigCounters);
+        
+        
+        FileSystem fs = FileSystem.get(conf);
+        
+        Path reportPath = portBindings.getOutput().get(REPORT_PORT_OUT_NAME);
+        
+        DataStore.create(avroReportParams, new FileSystemPath(fs, reportPath));
+        
+    }
+    
+    
+    //------------------------ PRIVATE --------------------------
+    
+    private Map<String, String> collectReportParameters(Map<String, String> parameters) {
+        
+        return parameters.entrySet().stream()
+                .filter(property -> property.getKey().startsWith(REPORT_PROPERTY_PREFIX))
+                .map(x -> Pair.of(x.getKey().substring(REPORT_PROPERTY_PREFIX.length()), x.getValue()))
+                .collect(Collectors.toMap(e -> e.getLeft(), e -> e.getRight()));
+        
+    }
+    
+    private List<ReportParam> convertToAvroReportParams(Map<String, String> reportParameters, PigCounters pigCounters) {
```

Comment:
Wouldn't it be better to have it in a separate service? It would be simpler and easier to test.